netket/experimental/hilbert/fermions_2nd.py

```diff
@@ -0,0 +1,138 @@
+# Copyright 2022 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, List, Union
+from collections.abc import Iterable
+import jax.numpy as jnp
+import numpy as np
+from fractions import Fraction
+
+from netket.hilbert.fock import Fock
+from netket.hilbert.tensor_hilbert import TensorHilbert
+from netket.hilbert.homogeneous import HomogeneousHilbert
+
+
+class SpinOrbitalFermions(HomogeneousHilbert):
+    r"""
+    Hilbert space for 2nd quantization fermions with spin `s` distributed among `n_orbital` orbitals.
+
+    The number of fermions can be fixed globally or fixed on a per spin projection.
+    """
+
+    def __init__(
+        self, n_orbitals: int, s: float = 0.0, n_fermions: Union[int, List[int]] = None
```

Comment:
I am still a bit uncomfortable with spin being default 0 (bosonic!), can't we set it to `None` instead?

Reply:
I understand. But the main fermionic property is simply the anti-commutation rules. In any framework you'll see 'spinless=True' as an option, so I don't think it's bad to allow to give spin=0. In any case, I can add the s=None as the default.

Reply:
does it make a difference in any way?

Reply:
code wise internally everything is the same, it's just nicer I think if s=None refers to spinless

Reply:
ok, I like it